app/src/androidTest/java/com/danielpriddle/drpnews/workers/GlideWorkerTest.kt

```diff
@@ -0,0 +1,51 @@
+package com.danielpriddle.drpnews.workers
+
+import android.content.Context
+import android.util.Log
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.work.Configuration
+import androidx.work.ListenableWorker.Result
+import androidx.work.WorkManager
+import androidx.work.impl.utils.SynchronousExecutor
+import androidx.work.testing.TestListenableWorkerBuilder
+import androidx.work.testing.WorkManagerTestInitHelper
+import androidx.work.workDataOf
+import com.bumptech.glide.Glide
+import org.junit.Assert
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import kotlin.reflect.typeOf
+
+class GlideWorkerTest {
+    private lateinit var workManager: WorkManager
+    private lateinit var context: Context
+    private lateinit var configuration: Configuration
+    private val IMAGE_URL = "https://images.wsj.net/im-646132/social"
+
+    @Before
+    fun setup() {
+        // Configure WorkManager
+        configuration = Configuration.Builder()
+            .setMinimumLoggingLevel(Log.DEBUG)
+            .setExecutor(SynchronousExecutor())
+            .build()
+
+        context = InstrumentationRegistry.getInstrumentation().targetContext
+        WorkManagerTestInitHelper.initializeTestWorkManager(context, configuration)
+        workManager = WorkManager.getInstance(context)
+    }
+
+    @Test
+    fun testGlideWork() {
+        val worker = TestListenableWorkerBuilder<GlideWorker>(
```

Comment:
Looks like this test fails because Glide is attempting to make an actual network fetch.  You can mock the Glide calls like this:

```kotlin
        mockkStatic(Glide::class)
        val mockGlideRequestManager = mockk<RequestManager>()
        val mockGlideRequestFileBuilder = mockk<RequestBuilder<File>>()
        val mockGlideFileFutureTarget = mockk<FutureTarget<File>>()
        val mockFile = mockk<File>()
        val imageFilePath = "/path/to/file"
        every { Glide.with(any<Context>()) } returns mockGlideRequestManager
        every { mockGlideRequestManager.asFile() } returns mockGlideRequestFileBuilder
        every { mockGlideRequestFileBuilder.load(IMAGE_URL) } returns mockGlideRequestFileBuilder
        every { mockGlideRequestFileBuilder.submit() } returns mockGlideFileFutureTarget
        every { mockGlideFileFutureTarget.get() } returns mockFile
        every { mockFile.path } returns imageFilePath
```

You'd also need to add the mockk-agent lib in for `androidTest` in your build.gradle:

```groovy
androidTestImplementation "io.mockk:mockk-agent:$mockk_version"
```

And then also add the following to your `packagingOptions` block in the build.gradle:

```groovy
jniLibs {
    useLegacyPackaging = true
}
```

A lot of mock setup admittedly 😅

Reply:
oh gross...

In hindset, I am not sure if I should've written this test. It feels like I am testing the Glide library itself, which the authors of Glide probably already did for me, rather than testing my worker/

Reply:
Technically it's good to have a test here for the worker itself, because you can mock the Glide response (that way you're not testing Glide itself) and ensure that the response from Glide comes through to the worker and the worker completes as expected.  But it is gross haha

Reply:
Fixed (I think) with [c2d7439](https://github.com/DPriddle96/DRPNews/pull/8/commits/c2d7439a47ffeeb4d9bc6dd3adf4bfb62768b041)